src/sqlfluff/core/parser/grammar/anyof.py

```diff
@@ -240,3 +267,10 @@ def __init__(self, *args, **kwargs):
             args[0] if len(args) == 1 else Sequence(*args),
             **kwargs,
         )
+
+
+class AnySetOf(AnyNumberOf):
+    """Match any number of the elements but each element can only be matched once."""
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, max_times_per_element=1, **kwargs)
```

Comment:
This just wraps the functionality in a convenient class for dialect contributors

Reply:
Nice!